torch/csrc/autograd/profiler.cpp

```diff
@@ -63,9 +63,29 @@ void pushRangeImpl(
     return;
   }
   if (state == ProfilerState::NVTX) {
-    if(sequence_nr >= 0) {
+    if(sequence_nr >= 0 || shapes.size() > 0) {
       std::stringstream s;
-      s << name.str() << msg << sequence_nr;
+      if(sequence_nr >= 0)
+        s << name.str() << msg << sequence_nr;
+      if(shapes.size() > 0) {
+        s << ", sizes = [";
+        for(int i = 0; i < shapes.size(); i++) {
```

Comment:
Could be there's a cleaner way to get the formatting right than these non-range-based for loops and if statements.  I'm ok with it as is but open to clever suggestions...

Reply:
You can probably kill the inner loop by casting it to an ArrayRef and then printing that (since ArrayRef has printing.) But if you don't feel like mucking with this code I don't mind.